### PR TITLE
swift: add version 5.7

### LINKED ISSUE #3844: [Request] Swift Compiler
## Package Request

It would be cool to have the Swift compiler as a package.

### Criteria

For a package to be acceptable in the Main bucket, it should be:

- [x] a non-GUI tool
- [x] a reasonably well-known and widely used developer tool (e.g. if it's a GitHub project, it should have at least 500 stars and 150 forks)
- [x] the latest stable version of the program
- [x] the full version i.e. not a trial version
- [x] a fairly standard install (e.g. uses a version-specific download URL, no elaborate pre/post install scripts)

### Information

**Name:** `swift`

**Description:** The toolchain for Apple's programming language Swift

**Homepage:** https://www.swift.org/

**Download link(s):** https://download.swift.org/swift-5.6.2-release/windows10/swift-5.6.2-RELEASE/swift-5.6.2-RELEASE-windows10.exe

**Some indication of popularity/repute:** 60k stars on GitHub (and generally pretty well-known): https://github.com/apple/swift

### PR DESCRIPTION
**Pull request by KrisSchneider**
Closes #3844

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/swift.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        "python": "main/python",
+        "vcredist": "extras/vcredist"
+    },
+    "notes": [
+        "To make Swift toolchain work, you must have Visual Studio and Windows SDK components installed.",
+        "Please read https://www.swift.org/getting-started/#support-files for Supporting Files installation guide.",
+        "For documentation on Swift, please click here https://www.swift.org/documentation/",
+        ""
+    ],
+    "depends": "dark",
```
Comment (rashil2000) [inline bucket/swift.json:16]:
```suggestion
```
Not explicitly required

File: bucket/swift.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "license": "Apache-2.0",
+    "suggest": {
+        "python": "main/python",
+        "vcredist": "extras/vcredist"
+    },
+    "notes": [
+        "To make Swift toolchain work, you must have Visual Studio and Windows SDK components installed.",
+        "Please read https://www.swift.org/getting-started/#support-files for Supporting Files installation guide.",
+        "For documentation on Swift, please click here https://www.swift.org/documentation/",
+        ""
```
Comment (rashil2000) [inline bucket/swift.json:14]:
```suggestion
        "Read https://www.swift.org/getting-started/#support-files for Supporting Files installation guide."
```

Link to package doc is not required.

File: bucket/swift.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    ],
+    "depends": "dark",
+    "url": "https://download.swift.org/swift-5.7-release/windows10/swift-5.7-RELEASE/swift-5.7-RELEASE-windows10.exe#/swiftsetup.exe",
+    "hash": "CC5580E347C492CF4F6C74E0E59D78938CB7B1484BA9A72CBD923C6EE6A4FE3D",
+    "pre_install": [
+        "Expand-DarkArchive \"$dir\\swiftsetup.exe\" \"$dir\\extract_folder\" -Removal",
+        "dir \"$dir\\extract_folder\\AttachedContainer\\*.msi\" | ForEach-Object {Expand-MsiArchive $_ \"$dir\" | Out-Null}",
```
Comment (rashil2000) [inline bucket/swift.json:21]:
```suggestion
        "Get-ChildItem \"$dir\\extract_folder\\AttachedContainer\\*.msi\" | ForEach-Object { Expand-MsiArchive $_ \"$dir\" | Out-Null }",
```

File: bucket/swift.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    "env_add_path": [
+        "Developer\\Toolchains\\unknown-Asserts-development.xctoolchain\\usr\\bin",
+        "icu-69.1\\usr\\bin",
+        "runtime-development\\usr\\bin"
+    ],
+    "env_set": {
+        "SDKROOT": "$dir\\Developer\\Platforms\\Windows.platform\\Developer\\SDKs\\Windows.sdk",
+        "DEVELOPER_DIR": "$dir\\Developer"
+    },
+    "pre_uninstall": "if (!(is_admin)) {error \"$app requires admin rights to $cmd\"; break}",
```
Comment (rashil2000) [inline bucket/swift.json:33]: Why is this required?

File: bucket/swift.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    "checkver": {
+        "github": "https://github.com/apple/swift/",
+        "regex": "releases/tag/swift-([\\d.]+)-RELEASE"
```
Comment (rashil2000) [inline bucket/swift.json:36]:
```suggestion
        "url": "https://www.swift.org/download/",
        "regex": "/swift-([\\d.]+)-RELEASE-windows10\\.exe"
```

Comment (KrisSchneider) [reply bucket/swift.json:33]: Why would I change this. If I use the swift website instead of GitHub, the checkver may not work.

Comment (rashil2000) [reply bucket/swift.json:33]: The change has been suggested only after testing it.

Comment (KrisSchneider) [reply bucket/swift.json:33]: Oh ok, good to know.

File: bucket/swift.json (hunk 4 above)
Comment (KrisSchneider) [reply bucket/swift.json:30]: I felt it was necessary. But if it's not needed, I can remove it.

Comment (rashil2000) [reply bucket/swift.json:30]: Please remove it

Comment (KrisSchneider) [reply bucket/swift.json:30]: Sure, I will do so in 15 mins.

Comment (KrisSchneider) [reply bucket/swift.json]: @rashil2000 it's done now

**Comment by KrisSchneider** [comment]
Should I reverse the manifest file back to a previous commit? Or are you going to merge this PR anyway despite the bot labelling this as `manifest-fix-needed`?

File: bucket/swift.json (hunk 5 above)
Comment (rashil2000) [reply bucket/swift.json]: The regex is correct: https://regex101.com/r/PMbSkC/1

There seems to be a problem with the encoding of response from `www.swift.org/download` webpage. 

We should revert to the GitHub one in that case.

Comment (KrisSchneider) [reply bucket/swift.json]: Sure, I can do that.

File: bucket/swift.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        "icu-69.1\\usr\\bin",
+        "runtime-development\\usr\\bin"
+    ],
+    "env_set": {
+        "SDKROOT": "$dir\\Developer\\Platforms\\Windows.platform\\Developer\\SDKs\\Windows.sdk",
+        "DEVELOPER_DIR": "$dir\\Developer"
+    },
+    "checkver": {
+        "url": "https://github.com/apple/swift/",
+        "regex": "/swift-([\\d.]+)-RELEASE-windows"
```
Comment (rashil2000) [inline bucket/swift.json:32]:
```suggestion
        "regex": "/swift-([\\d.]+)-RELEASE"
```

Comment (KrisSchneider) [reply bucket/swift.json:32]: I'll try that, thank you @rashil2000

**Comment by KrisSchneider** [comment]
The regex worked, thanks for the additional help and god bless you @rashil2000

**Comment by rashil2000** [comment]
Okay, so I tested this locally, and found out some things:
- all binaries are 64bit, so we'll have to use the architecture 64bit block
- `icu-69.1\\usr\\bin` does not include any executables, so we can leave it out of `env_add_path`
- `runtime-development\\usr\\bin` contains one executable, so we can use `bin` for it instead of `env_add_path`

**Comment by KrisSchneider** [comment]
> Okay, so I tested this locally, and found out some things:
> 
>     * all binaries are 64bit, so we'll have to use the architecture 64bit block
> 
>     * `icu-69.1\\usr\\bin` does not include any executables, so we can leave it out of `env_add_path`
> 
>     * `runtime-development\\usr\\bin` contains one executable, so we can use `bin` for it instead of `env_add_path`

Alright, it's done now.

**Comment by rashil2000** [comment]
Great, thanks for the effort @KrisSchneider!
